plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -82,17 +89,22 @@ export class PersonState {
     }
 
     async updateProperties(): Promise<Person | undefined> {
-        const createdPerson = await this.createPersonIfDistinctIdIsNew()
+        let person = this.person
+        let personCreated = false
+        if (!person) {
+            person = await this.createPersonIfDistinctIdIsNew()
```

Comment:
naming is slightly confusing as this can return undefined especially at the end return